test/functional/legacy/062_tab_pages_spec.lua

```diff
@@ -0,0 +1,330 @@
+-- Tests for tab pages
+
+local helpers = require('test.functional.helpers')
+local feed, insert, source = helpers.feed, helpers.insert, helpers.source
+local clear, execute, expect = helpers.clear, helpers.execute, helpers.expect
+
+describe('tab pages', function()
+  setup(clear)
+
+  it('is working', function()
+    insert([=[
+      Results:]=])
+
+    execute('lang C')
+
+    -- Simple test for opening and closing a tab page.
+    source([[
+      tabnew
+      let nr = tabpagenr()
+      q
+      call append(line('$'), 'tab page ' . nr)
+      unlet nr
+    ]])
```

Comment:
We can simplify this test by removing  "tab page 2" from the results and instead putting
```lua
execute('tabnew')
eq(2, eval('tabpagenr()'))
execute('q')
```
here, and similar for the other sections.